references/detection/transforms.py

```diff
@@ -326,3 +326,78 @@ def forward(
                 )
 
         return image, target
+
+
+class FixedSizeCrop(nn.Module):
+    def __init__(self, size, fill=0, padding_mode="constant"):
+        super().__init__()
+        size = tuple(T._setup_size(size, error_msg="Please provide only two dimensions (h, w) for size."))
+        self.crop_height = size[0]
+        self.crop_width = size[1]
+        self.fill = tuple(int(x) for x in fill)  # TODO: Fill is currently respected only on PIL. Apply tensor patch.
+        self.padding_mode = padding_mode
+
+    def _pad(self, img, target, padding):
+        # Taken from the functional_tensor.py pad
+        if isinstance(padding, int):
+            pad_left = pad_right = pad_top = pad_bottom = padding
+        elif len(padding) == 1:
+            pad_left = pad_right = pad_top = pad_bottom = padding[0]
+        elif len(padding) == 2:
+            pad_left = pad_right = padding[0]
+            pad_top = pad_bottom = padding[1]
+        else:
+            pad_left = padding[0]
+            pad_top = padding[1]
+            pad_right = padding[2]
+            pad_bottom = padding[3]
```

Comment:
Nit: in D2 they pad only in right/bottom direction:
- https://github.com/facebookresearch/detectron2/blob/9258799e4e72786edd67940872e0ed2c4387aac5/detectron2/data/transforms/augmentation_impl.py#L334-L345
- https://github.com/facebookresearch/fvcore/blob/279b7d093aa078847fe13fefcc2968280abe5297/fvcore/transforms/transform.py#L747-L767

Reply:
I don't understand, that's what I also do too.

This snippet as I mention above it taken from the `pad()` of `functional_tensor.py`. When this eventually gets refactored and ported to the new API we should consider making it a utility and share its functionality.

Reply:
We can ignore my comment as below you specify the padding as `[0, 0, p_right, p_bottom]`.